src/dodal/devices/tetramm.py

```diff
@@ -0,0 +1,273 @@
+    TwentyFourBits = "24 bits"
+
+
+class TetrammGeometry(str, Enum):
+    Diamond = "Diamond"
+    Square = "Square"
+
+
+class TetrammDriver(Device):
+    def __init__(
+        self,
+        prefix: str,
+        name: str = "",
+    ):
+        self._prefix = prefix
+        self.range = ad_rw(TetrammRange, prefix + "Range")
+        self.sample_time = ad_r(float, prefix + "SampleTime")
+
+        self.values_per_reading = ad_rw(int, prefix + "ValuesPerRead")
+        self.averaging_time = ad_rw(float, prefix + "AveragingTime")
+        self.to_average = ad_r(int, prefix + "NumAverage")
+        self.averaged = ad_r(int, prefix + "NumAveraged")
+
+        self.acquire = ad_rw(bool, prefix + "Acquire")
+
+        # this PV is special, for some reason it doesn't have a _RBV suffix...
+        self.overflows = epics_signal_r(int, prefix + "RingOverflows")
+
+        self.num_channels = ad_rw(TetrammChannels, prefix + "NumChannels")
+        self.resolution = ad_rw(TetrammResolution, prefix + "Resolution")
+        self.trigger_mode = ad_rw(TetrammTrigger, prefix + "TriggerMode")
+        self.bias = ad_rw(bool, prefix + "BiasState")
+        self.bias_volts = ad_rw(float, prefix + "BiasVoltage")
+        self.geometry = ad_rw(TetrammGeometry, prefix + "Geometry")
+        self.nd_attributes_file = epics_signal_rw(str, prefix + "NDAttributesFile")
+
+        super().__init__(name=name)
+
+
+class TetrammController(DetectorControl):
+    """Controller for a TetrAMM current monitor
+
+    Attributes:
+        base_sample_rate (int): Fixed in hardware
+
+    Args:
+        drv (TetrammDriver): A configured driver for the device
+        maximum_readings_per_frame (int): Maximum number of readings per frame: actual readings may be lower if higher frame rate is required
+        minimum_values_per_reading (int): Lower bound on the values that will be averaged to create a single reading
+        readings_per_frame (int): Actual number of readings per frame.
+
+    """
+
+    base_sample_rate: int = 100_000
+
+    def __init__(
+        self,
+        drv: TetrammDriver,
+        minimum_values_per_reading: int = 5,
+        maximum_readings_per_frame: int = 1_000,
+        readings_per_frame: int = 1_000,
+    ):
+        # TODO: Are any of these also fixed by hardware constraints?
+        self._drv = drv
+        self.maximum_readings_per_frame = maximum_readings_per_frame
+        self.minimum_values_per_reading = minimum_values_per_reading
+        self.readings_per_frame = readings_per_frame
+
+    def get_deadtime(self, exposure: float) -> float:
+        # 2 internal clock cycles. Best effort approximation
+        return 2 / self.base_sample_rate
+
+    async def arm(
+        self,
+        num: int,
+        trigger: DetectorTrigger,
+        exposure: float,
+    ) -> AsyncStatus:
+        if trigger not in {DetectorTrigger.edge_trigger, DetectorTrigger.constant_gate}:
+            raise ValueError("Only edge triggers are supported")
```

Comment:
The exception text doesn't seem to match the allowed cases: a constant_gate may be actually triggering off a rising or falling edge, but the message implies it can only support edge_trigger exactly?